e2e/_suites/ingest-manager/features/agent_endpoint_integration.feature

```diff
@@ -0,0 +1,38 @@
+@agent_endpoint_integration
+Feature: Fleet Mode Agent with Endpoint Integration
+  Scenarios for Agent to deploy Endpoint and sending data to Ingest Manager and Elasticsearch.
+
+@deploy-endpoint-with-agent
+Scenario: Deploy Agent and add Endpoint Integration and host shows in Security App
+  Given an agent is deployed to Fleet
+  When the "Endpoint" latest package version is installed successfully
+    And the "Endpoint" integration is "added" in the "default" configuration
+  Then the "default" configuration shows the "Endpoint" datasource added
+    And the host name is shown in the security app
+    
+@endpoint-policy-check
+Scenario: Deploy Endpoint and see policy return successful in Security App
+  Given an Endpoint is successfully deployed with Agent
+  Then the policy response will be listed in the security app
+
+@set-policy-and-check-changes
+Scenario: Deploy Endpoint and change default policy and verify in Security App
+  Given an Endpoint is successfully deployed with Agent
+  When the policy is updated to have malware in detect mode
```

Comment:
How can we create a policy so that it can be updated? We'd need both API callls: /POST and /PATCH

Reply:
step 1) to create the new config call POST to: 
api/ingest_manager/agent_configs?sys_monitoring=true

 - with a body of
`{"name":"e2e","description":"BDD Test","namespace":"default","monitoring_enabled":["logs","metrics"]}
`

to verify, we can query / poll & retry on:

api/ingest_manager/agent_configs?page=1&perPage=20&sortField=updated_at&sortOrder=desc&kuery=

and the response list element[0] shall have its 'name' attribute as "e2e".   
 - capture the 'id' field from this element[0] as the id of the config

screenshot:
<img width="330" alt="create-config" src="https://user-images.githubusercontent.com/12970373/90703596-2d1d8780-e25c-11ea-9259-ee0b03763857.png">



step 2) then POST to 
api/ingest_manager/package_configs
with a body of:
`{"name":"e2e-endpoint","description":"ep integration","namespace":"default","config_id":"710e4290-e25c-11ea-b22d-e91baaf6bb53","enabled":true,"output_id":"","inputs":[],"package":{"name":"endpoint","title":"Elastic Endpoint Security","version":"0.15.0"}}
`
where... the 710 is the ID of the config.

the name we desire 'e2e-endpoint' and a description 'ep integration'

the 'package' field is what we determined from the registry call... didn't we go thru that earlier?  maybe we did in the other pr going.  :)  the 'version' field should ideally be dynamic as it can change at any time and we'd ideally just have the test use the current version returned.

screenshot:
<img width="586" alt="modifiy-config-resp" src="https://user-images.githubusercontent.com/12970373/90703613-360e5900-e25c-11ea-8373-2b89bb23d61c.png">


step 3) then to update the Agent to have the cited config:
PUT /api/ingest_manager/fleet/agents/20728023-127b-47c4-bfc3-30ddbeda1124/reassign
with a body of:
`{"config_id":"710e4290-e25c-11ea-b22d-e91baaf6bb53"}
`
where 207... is the id of the Agent and 710... is the id of the newly created config

then to confirm, if we follow thru with a GET to the new ID as: api/ingest_manager/agent_configs/710e4290-e25c-11ea-b22d-e91baaf6bb53
the 'item' struct comes back with a kv pair of:
config_id: "710e4290-e25c-11ea-b22d-e91baaf6bb53"

screenshot:
<img width="498" alt="get-agent" src="https://user-images.githubusercontent.com/12970373/90703632-41fa1b00-e25c-11ea-91b5-f2bed3318ba9.png">